src/aspire/utils/matlab_compat.py

```diff
@@ -94,3 +94,21 @@ def __enter__(self):
     def __exit__(self, *args):
         if self.seed is not None:
             np.random.set_state(random_states.pop())
+
+
+def top_eigenvectors(v, sort_idx):
```

Comment:
This function has a very weird signature - the caller first needs to find out the eigenvalues/eigenvectors/eigenvalues (which are sorted anyway, or can be, by passing in appropriate flags to scipy - look at the `which` parameter), index into them, and pass you the truncated sort index. You're making the caller do a lot of work for very little gains.

Why not just have a `eigsh` wrapper function and pass all args/kwargs to the underlying scipy function, but then do the sign change to make it matlab-compatible? If there's a way to do this in the scipy function itself and totally eliminate this, that would be ideal.

Reply:
I believe this is not the only function from scipy or numpy with the similar problem. The last one I encounter is scipy.linalg.eigh_tridiagonal for PSWF2D basis class. I initial thinking is letting this `top_eigenvectors` util function to fix the sign problem of eigenvectors for other scipy functions too instead of  developing all wrapper functions. But let me know if you prefer to individual wrapper functions.

Reply:
Changed to use wrapper function in commit ec730a2.